src/index.ts

```diff
@@ -146,11 +145,21 @@ class Tx {
     this.conn = conn
   }
 
-  async execute(
+  async execute<T = Record<string, any>>(
+    query: string,
+    args?: ExecuteArgs,
+    options?: { as?: 'object'; cast?: Cast }
+  ): Promise<ExecutedQuery<T>>
+  async execute<T = any[]>(
+    query: string,
+    args: ExecuteArgs,
+    options: { as: 'array'; cast?: Cast }
+  ): Promise<ExecutedQuery<T>>
+  async execute<T = any>(
     query: string,
     args: ExecuteArgs = null,
-    options: ExecuteOptions = defaultExecuteOptions
-  ): Promise<ExecutedQuery> {
+    options: any = { as: 'object' }
+  ): Promise<ExecutedQuery<any>> {
```

Comment:
```suggestion
  ): Promise<ExecutedQuery<T>> {
```

Do we need to propagate this down to here?